src/hotspot/share/gc/serial/cardTableRS.cpp

```diff
@@ -601,20 +345,188 @@ void CardTableRS::verify() {
 CardTableRS::CardTableRS(MemRegion whole_heap) :
   CardTable(whole_heap) { }
 
+// Implemented word-iteration to skip long consecutive clean cards.
+CardTable::CardValue* CardTableRS::find_first_dirty_card(CardValue* const start_card,
+                                                         CardValue* const end_card) {
+  using Word = uintptr_t;
+
+  static_assert(clean_card_val() == (CardValue)-1, "inv");
+  constexpr Word clean_word = (Word)CardTable::clean_card_row_val();
+
+  CardValue* i_card = start_card;
```

Comment:
What does the "i_" prefix here mean?  Something like "cur_card" might be a better name.

Reply:
I don't think I would have ever guessed that, which suggests it might not be an ideal naming choice.

Reply:
My concern regarding `cur_card` is that there is no contrasting `prev_` or `next_` in this context. The sole message I want to convey here is that we are iterating through cards. Using the conventional loop variable `i` plus `card` seems like the most natural approach, in my opinion. It is concise, so it doesn't introduce much overhead despite its frequent occurrence in the context. Therefore, I prefer the current name. By the way, the same style and convention are also employed in other methods in this PR. Additionally, similar naming conventions can be found in the already merged code of Parallel (`PSCardTable`) and G1 (`G1ScanHRForRegionClosure`).

Reply:
I think "cur" is a frequently used naming convention for iteration, irrespective of there being related
"prev" or "next".  I don't like the similar naming in those other places you mention either.  Like I said,
"i" usually indicates an integral/index variable, and using it otherwise is (at least to me) confusing.

Reply:
Given that there is now a strong objection to this naming style, and I did not really like the `i_` prefix in the other CRs either but did not speak up, I suggest we just use the existing common `cur_` prefix here. From my point of view there is no relation of `cur_` to `prev`/`next` either.

The other uses can be fixed up separately.